docs/conventions.md

```diff
@@ -0,0 +1,45 @@
+# Conventions
+This document plans to illustrate the expectations set upon developers contributing to this project. It is intended for such conventions to be relatively lightweight and not overly restrictive, but to provide a baseline for how the project is structured and maintained.
+
+## Language
+All code is to be written in `TypeScript` unless there is a specific reason to use another language. This is to ensure that the codebase is consistent and that all developers can understand the codebase and sing off the same hymn sheet.
+
+## Commit messages
+This project is following the [Conventional Commits Standard](https://www.conventionalcommits.org/en/v1.0.0/#summary). This is to ensure that commit messages maintain a level of consistency describing:
+
+1. What kind of change it is, e.g. new feature or bugfix
+2. What component they affect
+3. A brief description of the change
+
+- `feat(web): Add new feature to the web component`
+- `fix(parser): Fix bug in the parser component`
+- `chore(repo): Update README.md`
+- `docs(repo): Update conventions.md`
+- `ci(repo): Update CI configuration`
+
+See [Versioning](./versioning.md) on why this is important and why we use it in practice.
+
+## Code standards
+**ALL** code despite language is to be written in the following format:
+
+- 2 **spaces** for indentation
+- No trailing whitespace
+- Must have LF line endings
+- Must have a newline at the end of the file
+- Must comply with the ESLint specifications (typescript)
+
+To aid this, developers can run `pnpm lint` in any component which will run ESLint checks and fix them. You must do this before commiting or the CI will block the PR via a status check.
+
+## Branching strategy
+There is none. We are not enforcing any form of branching naming strategy. 
```

Comment:
I'd like to propose we add that we typically run trunk based strategy. Create a feature branch, do the work, merge it back to main. Like a navy seal, get in there, do the work, get out, and if your job is done well, very few people know you are there (e.g. the tests pass and your feature is useful). 

Branch naming isn't as important as stopping people from creating their own long-lived dev branch

Reply:
That's pretty much what we are doing?

Reply:
Maybe it's just the title that is throwing me off - maybe it should be named `branch naming strategy`?

Reply:
Yes that's probably accurate